arbitrum-docs/stylus/how-tos/debugging-stylus-tx.mdx

```diff
@@ -0,0 +1,174 @@
+---
+title: 'How to debug Stylus transactions using Cargo Stylus Replay'
+description: 'This how-to explains how to perform trace calls and use GDB to replay and debug Stylus transactions, providing detailed analysis and troubleshooting.'
+author: mahsamoosavi
+sme: mahsamoosavi
+target_audience: 'Developers deploying smart contracts using Stylus'
+content_type: how-to
+sidebar_position: 2
+---
+
+import PublicPreviewBannerPartial from '../partials/_stylus-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+Debugging smart contracts can be challenging, especially when dealing with complex transactions. The `cargo-stylus-replay` crate simplifies the debugging process by allowing developers to replay Stylus transactions. This tool leverages GDB to provide an interactive debugging experience, enabling developers to set breakpoints, inspect state changes, and trace the execution flow step-by-step. This capability is crucial for identifying and resolving issues, ensuring that smart contracts function correctly and efficiently.
+
+### Overview
+
+Cargo Stylus is a tool designed to simplify the development and debugging process for smart contracts written in Rust for the Stylus execution environment. One of its powerful features is the `cargo-stylus-replay` crate, which provides essential functionalities for developers:
+
+1. **Trace calls**: Perform trace calls against Stylus transactions using Ethereum nodes' `debug_traceTransaction` RPC. This feature enables developers to analyze the execution flow and state changes of their transactions in a detailed manner.
+2. **Debugging with GDB**: Replay and debug the execution of a Stylus transaction using the GNU Debugger (GDB). This allows developers to set breakpoints, inspect variables, and step through the transaction execution line by line, providing an in-depth understanding of the transaction's behavior.
+
+### Replaying transactions
+
+**Note:** Currently, this has only been tested on Linux, x86 systems with GDB installed.
+
+#### Requirements
+
+- **Rust** (version 1.77 or higher)
+- **Crates**: `cargo-stylus`, `cargo-stylus-replay`, `cargo-stylus-check`
+- **GNU Debugger (GDB)**
+- **[Cast](https://book.getfoundry.sh/cast/)** (an Ethereum CLI tool)
+- **Local Arbitrum Sepolia node** with tracing endpoints enabled or a local Stylus testnet v2 node
```

Comment:
Is v2 deprecated now? I forget 😁. It might make more sense to mention the testnode here since that should be easy to test against

Reply:
agreed, I'll change it to testnode